qa/tasks/scripts/ses_rack_dc_region_unavailability.sh

```diff
@@ -0,0 +1,249 @@
+do
+    ceph osd crush move $osd_node rack=rack2
+done
+ 
+
+
+# region2
+split_nodes ${#hosts[@]}
+
+# nodes for region2
+for region2 in $(seq 1 $second_part)
+do
+    region2_hosts+=(${hosts[-$region2]})
+done
+
+# split region2 nodes to racks
+split_nodes ${#region2_hosts[@]}
+
+# nodes for rack3 in region2
+for rack3 in $(seq 0 $(($first_part - 1)))
+do
+    rack3_hosts+=(${region2_hosts[$rack3]})
+done
+
+# nodes for rack4 in region2
+for rack4 in $(seq 1 $second_part)
+do
+    rack4_hosts+=(${region2_hosts[-$rack4]})
+done
+
+for osd_node in ${rack3_hosts[@]}
+do
+    ceph osd crush move $osd_node rack=rack3
+done
+ 
+for osd_node in ${rack4_hosts[@]}
+do
+    ceph osd crush move $osd_node rack=rack4
+done
+ 
+# bring down rack
+for node2fail in ${rack4_hosts[@]}
+do
+    salt ${node2fail}.teuthology cmd.run "iptables -I OUTPUT -d localhost -j ACCEPT"
+    salt ${node2fail}.teuthology cmd.run "iptables -I OUTPUT -d $master -j ACCEPT"
+    salt ${node2fail}.teuthology cmd.run "iptables -I INPUT -s localhost -j ACCEPT"
+    salt ${node2fail}.teuthology cmd.run "iptables -I INPUT -s $master -j ACCEPT"
+    salt ${node2fail}.teuthology cmd.run "iptables -P INPUT DROP"
+    salt ${node2fail}.teuthology cmd.run "iptables -P OUTPUT DROP"
+done
+
+until ceph -s | grep ".* rack.* down"
+do
+    sleep 30
+done 
+
+ceph -s
+
+ceph osd tree
+
+# bring rack up
+for node2fail in ${rack4_hosts[@]}
+do
+    salt ${node2fail}.teuthology cmd.run "iptables -P INPUT ACCEPT"
+    salt ${node2fail}.teuthology cmd.run "iptables -P OUTPUT ACCEPT"
+    salt ${node2fail}.teuthology cmd.run "iptables -F"
+done
+
+cluster_health
+
+### DC failure
+ceph osd crush add-bucket dc1 datacenter
+ceph osd crush add-bucket dc2 datacenter
+ceph osd crush move dc1 root=$root_name
+ceph osd crush move dc2 root=$root_name
+ceph osd crush move rack1 datacenter=dc1
+ceph osd crush move rack2 datacenter=dc1
+ceph osd crush move rack3 datacenter=dc2
+ceph osd crush move rack4 datacenter=dc2
+
+dc1_nodes=(${rack1_hosts[@]} ${rack2_hosts[@]})
+dc2_nodes=(${rack3_hosts[@]} ${rack4_hosts[@]})
+
+# bringing down DC
+for node2fail in ${dc1_nodes[@]}
+do
+    salt ${node2fail}.teuthology cmd.run "iptables -I OUTPUT -d localhost -j ACCEPT"
+    salt ${node2fail}.teuthology cmd.run "iptables -I OUTPUT -d $master -j ACCEPT"
+    salt ${node2fail}.teuthology cmd.run "iptables -I INPUT -s localhost -j ACCEPT"
+    salt ${node2fail}.teuthology cmd.run "iptables -I INPUT -s $master -j ACCEPT"
+    salt ${node2fail}.teuthology cmd.run "iptables -P INPUT DROP"
+    salt ${node2fail}.teuthology cmd.run "iptables -P OUTPUT DROP"
+done
+
+until ceph -s | grep ".* datacenter.* down"
+do 
+    sleep 30
+done
+
+ceph -s 
+
+ceph osd tree
+
+# bring DC up
+for node2fail in ${dc1_nodes[@]}
+do
+    salt ${node2fail}.teuthology cmd.run "iptables -P INPUT ACCEPT"
+    salt ${node2fail}.teuthology cmd.run "iptables -P OUTPUT ACCEPT"
+    salt ${node2fail}.teuthology cmd.run "iptables -F"
+done
```

Comment:
Here and below it is the same algorithms used, could you make a function to avoid a code duplication?